apps/site/assets/ts/hooks/useFetch.ts

```diff
@@ -0,0 +1,27 @@
+import useSWR from "swr";
+import { fetchJsonOrThrow } from "../helpers/fetch-json";
+import { FetchState, FetchStatus } from "../helpers/use-fetch";
+
+/**
+ * Generic helper function for fetching JSON from a URL. Specify the expected
+ * return type as the type parameter, e.g
+ * `useFetch<Schedule[]>("/api/schedules/or/something")`
+ * @param {string | null} backendUrl - The path, usually starting with /api/,
+ * that our backend will respond with data from. If this value is null, useSWR's
+ * behavior will in effect disable the fetch, and the returned data will simply
+ * be undefined.
+ * @returns {FetchState<T>} An object containing the fetch status, and
+ * (optionally) the fetched data.
+ */
+const useFetch = <T>(backendPath: string | null): FetchState<T> => {
```

Comment:
Could this be named something else?  We already have a function called `useFetch` from the `helpers/use-fetch` file.  It would be nice to avoid name collisions to help with readability.

Reply:
I really missed that that file has, among other things, it's old overlapping `useFetch` 🤦🏼 ... yeah, will resolve.

Reply:
`helpers/use-fetch` is only used in two spots, so I think I'm gonna refactor those and make the `helpers/use-fetch` file obsolete.